src/components/BandInput.js

```diff
@@ -1,38 +1,33 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 
-class BandInput extends Component {
+const BandInput = ({ addBand }) => {
+  const [bandName, setBandName] = useState("");
 
-  state = {
-    bandName: ''
-  }
+  const handleChange = e => setBandName(e.target.value);
 
-  handleOnChange(event) {
-    this.setState({
-      bandName: event.target.value,
-    });
-  }
+  const handleSubmit = e => {
+    e.preventDefault();
 
-  handleOnSubmit(event) {
-    event.preventDefault();
-    this.props.addBand(this.state.bandName);
-    this.setState({
-      bandName: '',
-    });
-  }
+    if (!bandName || bandName === "Phish") {
+      alert("Please enter a band name");
+      return;
+    }
```

Comment:
`handleSubmit` is a fine time to do some validation here. This is a naive example, but let's make sure there is actual input before we add anything, and for good measure let's make sure nobody tried to submit _Phish_. 🤮 

Real world examples would probably have you cast this to lowercase (and change cassing presentationally, ie css) and looking at the `bands` array to ensure we're not duplicating something in the list. That's a little advanced for this, so we'll just keep it playful.

Reply:
I concur.